internal/daemon/daemon.go

```diff
@@ -419,3 +427,59 @@ func (d *Daemon) runSideManager(mgr SideManager, identifier string, managerCtx c
 	}
 	return nil
 }
+
+// updateNodeLabels updates the node labels based on the detected DPUs. This is done independently since labels can be
+// removed or modified accidently. We always want this label to be set/deleted properly.
+func (d *Daemon) updateNodeLabels() error {
```

Comment:
We're going back to labels instead of DPU CR? Not clear why

Reply:
You can't use the DPU CR as a node selector. So we need both to be able to schedule nodes properly in a one cluster deployment.

Reply:
This is done in code. Which you can take values from any resource and apply to any other resource. But this doesn't allow you to create workloads that you want to run on either the hosts or the dpu externally.

This is for workload pods that the user creates and that we have no control over.